addons/base_import/static/src/additionalFileManager.js

```diff
@@ -0,0 +1,93 @@
+import { checkFileSize } from "@web/core/utils/files";
+
+export class AdditionalFileManager {
+    constructor(resModel, fields, context, options, orm, notificationService, maxBatchSize) {
+        this.resModel = resModel;
+        this.fields = [".id", ...fields];
+
+        this.context = context;
+        this.options = options;
+        this.orm = orm;
+        this.notificationService = notificationService;
+        this.maxBatchSize = maxBatchSize * 0.95; // 0.95 for not calculated payload overhead
+
+        this.dataToSend = {};
+        this.lastSend = Date.now();
+        this.delayAfterEachBatch = this.options.delayAfterEachBatch * 1000;
+    }
+
+    async addFile(id, field, file) {
+        let data = await this._readFile(file);
+        if (typeof data === "string" && data.startsWith("data:")) {
+            // Remove data:image/*;base64,
+            data = data.split(",")[1];
+        }
+        const dataSize = data.length;
+        if (!checkFileSize(dataSize, this.notificationService)) {
+            return;
+        }
+
+        if (this.getCurrentSize() + dataSize >= this.maxBatchSize) {
+            await this._send();
+        }
+        if (!(id in this.dataToSend)) {
+            this.dataToSend[id] = Array(this.fields.length);
+            this.dataToSend[id][0] = id;
+        }
+        const indexOfField = this.fields.indexOf(field, 1);
+        this.dataToSend[id][indexOfField] = data;
+    }
+
+    async sendLastPayload() {
+        if (Object.keys(this.dataToSend).length > 0) {
+            await this._send();
+        }
+    }
+
+    async _send() {
+        const now = Date.now();
+        if (this.lastSend + this.delayAfterEachBatch <= now) {
+            await new Promise((resolve, reject) => {
+                setTimeout(resolve, this.delayAfterEachBatch);
+            });
+        }
+        try {
+            const data = Object.values(this.dataToSend);
+            const res = await this.orm.silent.call(this.resModel, "load", [], {
+                fields: this.fields,
+                data,
+                context: {
+                    ...this.context,
+                    tracking_disable: this.options.tracking_disable,
+                    import_file: true,
+                    name_create_enabled_fields: this.options.name_create_enabled_fields || {},
+                    import_set_empty_fields: this.options.import_set_empty_fields || [],
+                    import_skip_records: this.options.import_skip_records || [],
+                },
+            });
+            this.dataToSend = {};
+            return res;
+        } catch (error) {
+            console.error(error);
+            // This pattern isn't optimal but it is need to have
+            // similar behaviours as in legacy. That is, catching
```

Comment:
legacy? What is legacy?

Reply:
Logic copied from `_callImport` in the `import_model.js`.
Seems strange also to me but not expert of all legacy code.
If not needed we can remove it